src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,27 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        try (Session session = factory.openSession()) {
+            Transaction transaction = session.beginTransaction();
+            try {
+                session.persist(country);
+                transaction.commit();
+            } catch (Exception e) {
+                if (transaction != null) {
+                    transaction.rollback();
+                }
+                throw new DataProcessingException(
+                        "Can not to add country with name: " + country.getName());
+            }
+        }
+        return country;
     }
 
     @Override
     public Optional<Country> get(Long id) {
-        return null;
+        Session session = factory.openSession();
+        Country country = session.get(Country.class, id);
+        session.close();
+        return Optional.ofNullable(country);
     }
```

Comment:
Please use try with resources here. Also, you can use this: Optional.ofNullable(session.get(Country.class, id))